docs/api/MockAgent.md

```diff
@@ -38,6 +38,14 @@ const agent = new Agent()
 const mockAgent = new MockAgent({ agent })
 ```
 
+### Example - MockAgent instantiation with missed request callback
+
+```js
+import { MockAgent } from 'undici'
+
+const mockAgent = new MockAgent({ onMissedRequest: (error) => console.log(`A request wasn't handled: ${error.message}`) })
```

Comment:
I like the idea pretty much, but not sure if it should be applied to the `Agent`. Have you considered the interceptor instead?

In that way, I imagine a usage like:

```js
mockPool.intercept({
  path: '/justatest?hello=there&see=ya',
  method: 'GET'
}).reply(200, { wont: 'match' }, {
  headers: { 'content-type': 'application/json' }
// being response the object passed through the intercept
}).onNotMatched((request, response) => console.log(request, response))
```

Reply:
I think I like this usage instead as well. Any potential downsides to this syntax?

Reply:
I prefer this actually.